File: atcoder/abc311A_first_abc.cpp
// Vicfred
// https://atcoder.jp/contests/abc311/tasks/abc311_a
// implementation
#include <iostream>
#include <set>
#include <string>

using namespace std;

int main() {
  int n;
  cin >> n;
  string s;
  cin >> s;
  set<char> seen;
  int ans = 0;
  while(seen.size() < 3) {
    seen.insert(s[ans++]);
  }
  cout << ans << endl;
  return 0;
}

File: atcoder/abc311a_first_abc_.hs
-- vicfred
-- https://atcoder.jp/contests/abc311/tasks/abc311_a
-- implementation
import Data.Set qualified as Set
main :: IO ()
main = do
  _ <- getLine
  s <- getLine
  print $ f s Set.empty 1

f :: String -> Set.Set Char -> Int -> Int
f [] _ _ = 0
f (x: xs) seen idx
  | Set.size newSeen == 3 = idx
  | otherwise = f xs newSeen (idx + 1)
  where
    newSeen = Set.insert x seen

